lib/auth/auth_test.go

```diff
@@ -1104,6 +1104,11 @@ func TestSAMLConnectorCRUDEventsEmitted(t *testing.T) {
 	})
 	require.NoError(t, err)
 
+	role, err := types.NewRole("dummy", types.RoleSpecV5{})
```

Comment:
The role does need to be registered due to the changes I made. It used to be that validation was done in the grpc server, but has now moved to the auth server. So the tests used to work without the role because the saml connector was not validated completely by the auth server, but now is. Some recent changes added the role checking, but that role checking is not always done because it needs a RoleGetter (auth.Server or interface to) and that is not available in all places.

I'll comment appropriately and deduplicate the strings as you suggest.